### ISSUE TITLE
Feature request: hide jailbreak for specific app

### ISSUE THREAD

**Issue by missing233** [opa334/Dopamine#531]
Is it possible to to hide the jailbreak for a specific app?
I installed the SMBC Bank app and tried using Choicy, DopamineBlackList, but none of them worked. Only Dopamine's hidejailbreak feature works on it, but it's such a hassle to do every time (when the jailbreak is restored, the Sileo icon pops up in a different place, and its permissions get reset).

**Comment by opa334** [comment]
not possible

**Comment by missing233** [comment]
> not possible

So is it possible to keep the Sileo icon in the correct place and prevent permissions from being reset when the jailbreak restored?

**Comment by opa334** [comment]
> > not possible
> 
> So is it possible to keep the Sileo icon in the correct place and prevent permissions from being reset when the jailbreak restored?

Not really as it being removed is important to prevent apps from detecting it being installed. It's a feature, not a bug.

**Comment by EolnMsuk** [comment]
TLDR: It's curious that my bank app can detect the jailbreak even after I reboot my iPhone, yet dopamine is able to hide the jailbreak from said bank app while not in a jailbroken state.

1) I use California Coast Credit Union, even if I reboot my iPhone into a non-jailbroken state, the app will not let me get to the sign in screen, but if I open Dopamine and enable the hide my jailbreak feature, I can open the bank app. But if I am jailbroken, and it's not hidden, and I reboot into non-jailbroken state, it's still won't let me open the bank app, the only way I can open it while non-jailbroken is to open the Dopamine app and hide the jailbreak even though I'm not currently jailbroken. This setting saves even if I reboot my iPhone from a jailbroken state or from a non-jailbroken state.

2) For some reason (while not jailbroken) I can only switch the hide my jailbreak feature on once and then it disappears and I can't disable the hide my jailbreak anymore till I actually jailbreak again... rebooting keeps the same hidden feature enabled (it saves the setting) and I can't turn it back to the normal setting till I jailbreak again - at which point it has turned it off (normal).

**Comment by missing233** [comment]
> TLDR: It's curious that my bank app can detect the jailbreak even after I reboot my iPhone, yet dopamine is able to hide the jailbreak from said bank app while not in a jailbroken state.
> 
> 1. I use California Coast Credit Union, even if I reboot my iPhone into a non-jailbroken state, the app will not let me get to the sign in screen, but if I open Dopamine and enable the hide my jailbreak feature, I can open the bank app. But if I am jailbroken, and it's not hidden, and I reboot into non-jailbroken state, it's still won't let me open the bank app, the only way I can open it while non-jailbroken is to open the Dopamine app and hide the jailbreak even though I'm not currently jailbroken. This setting saves even if I reboot my iPhone from a jailbroken state or from a non-jailbroken state.
> 2. For some reason (while not jailbroken) I can only switch the hide my jailbreak feature on once and then it disappears and I can't disable the hide my jailbreak anymore till I actually jailbreak again... rebooting keeps the same hidden feature enabled (it saves the setting) and I can't turn it back to the normal setting till I jailbreak again - at which point it has turned it off (normal).

try Dopamine-roothide

**Comment by EolnMsuk** [comment]
What are the limitations of roothide
